robot/rover/internal_comms/README.md

```diff
@@ -16,5 +16,13 @@ The class also contains a virtual function that must be implemented. That functi
 Read the header and source files to understand what is happening. It is short and somewhat well documented. 
 The process is as follows:
 1. In the teensy loop, call the `readCommand` method. This will trickle down into your implementation of `executeCommand` if the command is valid.
-2. Call `sendMessage` in your loop. It will check the message queue and send a message if the transceiver pin is enabled. 
-3. Before sending anything, you must create a message struct using `createMessage` and you must enqueue it using `queueMessage`.
+2. Witin `executeCommand`, you must create a message struct using `createMessage`.
+3. If the commandID from the `executeCommand` is valid, add the created message to a queue using `sendMessage` within the case statement.  
+### Note ###
+This queue stores all messages which were sent, in case the message is received, but unable to be executed due to the UART pin being disabled. A UART pin can be disabled for a particular module if it is being used by a different module at a current moment. With the message added, then the latest message added to the queue is attempted to be sent. This is because the queue of messages must start by sending the latest message, and not the most recent! If this message is the only message in the queue, it is added, sent and removed from the queue.
+
+4. After the message is added to the queue, call the associated .ino function depending on the commandID
+5. Within the function for the specific command, call `sendMessage` in your loop.  
+### Note ###
+`sendMessage` will check the message queue, send a message if the transceiver pin is enabled, and then pop the message out of the queue. If `sendMessage` fails, the message will remain in the queue to be performed later. The queue's purpose is to keep track of previous messages, so that they can be performed once the serial pin is enabled for the particular module. 
```

Comment:
more like
the queue is a backlog of messages that will be sent once the transceiver pin gets enabled

Reply:
I'll reword it.

Reply:
sendMessage is overloaded so 
sendMessage() will check transmit pin and pop/send the first item from the queue.
sendMessage(Message&) will push the message to the queue and then run sendMessage()

IDK if that is what you meant

Reply:
I changed this btw, lmk if it makes more sense